R/package.R

```diff
@@ -113,15 +113,31 @@ initialize_python <- function(required_module = NULL, use_environment = NULL) {
   # set as appropriate)
   if (nzchar(config$virtualenv))
     Sys.setenv(VIRTUAL_ENV = config$virtualenv)
+  
+  # set R_SESSION_INITIALIZED flag (used by rpy2)
+  curr_session_env <- Sys.getenv("R_SESSION_INITIALIZED", unset = NA)
+  Sys.setenv(R_SESSION_INITIALIZED = sprintf('PID=%s:NAME="reticulate"', Sys.getpid()))
 
```

Comment:
I wonder if we could manage the failure case with an `on.exit()`. Something like this:

```r
on.exit({
  if (!config$available) {
    if (is.na(curr_session_env))
      Sys.unsetenv("R_SESSION_INITIALIZED")
    else 
      Sys.setenv(R_SESSION_INITIALIZED = curr_session_env)
  }
}, add = TRUE)
```

That way we wouldn't have to mess with the control flow below at all.

Reply:
I thought about but the `on.exit()` would have to add the handle the case of a successful initialization of Python and it did not seem much of a gain since the control of flow is in the `on.exit()` anyway.

Either of the two is a stylistic choice, I'd say. The PR's main purpose is to have the environment variable set before Python is initialized, I am happy with any way to handle failure during initialization you like :-) .

Reply:
Okay, I'm good with your implementation then.